lib/lints/avoid_unnecessary_type_assertions/avoid_unnecessary_type_assertions_fix.dart

```diff
@@ -0,0 +1,50 @@
+part of 'avoid_unnecessary_type_assertions_rule.dart';
+
+/// A Quick fix for `avoid-unnecessary-type-assertions` rule
+/// Suggests to remove unnecessary assertions
+class _UnnecessaryTypeAssertionsFix extends DartFix {
+  @override
+  void run(
+    CustomLintResolver resolver,
+    ChangeReporter reporter,
+    CustomLintContext context,
+    AnalysisError analysisError,
+    List<AnalysisError> others,
+  ) {
+    context.registry.addIsExpression((node) {
+      if (analysisError.sourceRange.intersects(node.sourceRange)) {
+        _addDeletion(reporter, 'is', node, node.isOperator.offset);
+      }
+    });
+
+    context.registry.addMethodInvocation((node) {
+      if (analysisError.sourceRange.intersects(node.sourceRange)) {
+        _addDeletion(
+          reporter,
+          'whereType',
```

Comment:
Don't you have it defined as a constant elsewhere?
Let's reuse a single definition instead of hardcoding strings.